src/sharing/ShareWrapper.js

```diff
@@ -157,13 +154,34 @@ class ShareWrapperInner extends React.PureComponent<Props, State> {
       return;
     }
 
-    this.setSending();
+    this.setState({ sending: true });
     showToast(_('Sending message…'));
     if (sharedData.type === 'file') {
       const { files } = this.state;
       for (let i = 0; i < files.length; i++) {
-        const response = await api.uploadFile(auth, files[i].url, files[i].name);
-        messageToSend += `\n[${files[i].name}](${response.uri})\n`;
+        const { url, name: fileName } = files[i];
+        try {
+          const response = await api.uploadFile(auth, url, fileName);
+          messageToSend += `\n[${fileName}](${response.uri})\n`;
+        } catch (errorIllTyped) {
+          const error: mixed = errorIllTyped; // https://github.com/facebook/flow/issues/2470
+
+          if (!(error instanceof Error)) {
+            logging.error('ShareWrapper: Unexpected non-error thrown');
+            return;
+          }
```

Comment:
If this somehow happens, seems best to still give the user feedback. Just won't have a message for them.